lib/srv/discovery/fetchers/aws-sync/s3_test.go

```diff
@@ -0,0 +1,198 @@
+	)
+	var (
+		regions = []string{"eu-west-1"}
+	)
+
+	tests := []struct {
+		name string
+		want *Resources
+	}{
+		{
+			name: "poll s3",
+			want: &Resources{
+				S3Buckets: []*accessgraphv1alpha.AWSS3BucketV1{
+					{
+						Name:           bucketName,
+						AccountId:      accountID,
+						PolicyDocument: []byte("policy"),
+						IsPublic:       true,
+						Acls: []*accessgraphv1alpha.AWSS3BucketACL{
+							{
+								Grantee: &accessgraphv1alpha.AWSS3BucketACLGrantee{
+									Id: "id",
+								},
+								Permission: "READ",
+							},
+						},
+						Tags: []*accessgraphv1alpha.AWSTag{
+							{
+								Key:   "tag",
+								Value: strPtrToWrapper(aws.String("val")),
+							},
+						},
+					},
+					{
+						Name:           otherBucketName,
+						AccountId:      accountID,
+						PolicyDocument: []byte("otherPolicy"),
+						IsPublic:       false,
+						Acls: []*accessgraphv1alpha.AWSS3BucketACL{
+							{
+								Grantee: &accessgraphv1alpha.AWSS3BucketACLGrantee{
+									Id: "id",
+								},
+								Permission: "READ",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockedClients := &cloud.TestCloudClients{
+				S3: &mocks.S3Mock{
+					Buckets: s3Buckets(bucketName, otherBucketName),
+					BucketPolicy: map[string]string{
+						bucketName:      "policy",
+						otherBucketName: "otherPolicy",
+					},
+					BucketPolicyStatus: map[string]*s3.PolicyStatus{
+						bucketName: {
+							IsPublic: aws.Bool(true),
+						},
+						otherBucketName: {
+							IsPublic: aws.Bool(false),
+						},
+					},
+					BucketACL: map[string][]*s3.Grant{
+						bucketName: {
+							{
+								Grantee: &s3.Grantee{
+									ID: aws.String("id"),
+								},
+								Permission: aws.String("READ"),
+							},
+						},
+						otherBucketName: {
+							{
+								Grantee: &s3.Grantee{
+									ID: aws.String("id"),
+								},
+								Permission: aws.String("READ"),
+							},
+						},
+					},
+					BucketTags: map[string][]*s3.Tag{
+						bucketName: {
+							{
+								Key:   aws.String("tag"),
+								Value: aws.String("val"),
+							},
+						},
+					},
+				},
+			}
+
+			var (
+				errs []error
+				mu   sync.Mutex
+			)
+
+			collectErr := func(err error) {
+				mu.Lock()
+				defer mu.Unlock()
+				errs = append(errs, err)
+			}
+			a := &awsFetcher{
+				Config: Config{
+					AccountID:    accountID,
+					CloudClients: mockedClients,
+					Regions:      regions,
+					Integration:  accountID,
+				},
+			}
+			result := &Resources{}
+			execFunc := a.pollAWSS3Buckets(context.Background(), result, collectErr)
+			require.NoError(t, execFunc())
+			require.NoError(t, trace.NewAggregate(errs...))
+
+			sort.Slice(result.S3Buckets, func(i, j int) bool {
+				return result.S3Buckets[i].Name < result.S3Buckets[j].Name
+			})
+			sort.Slice(tt.want.S3Buckets, func(i, j int) bool {
+				return result.S3Buckets[i].Name < result.S3Buckets[j].Name
```

Comment:
```suggestion
				return tt.want.S3Buckets[i].Name < tt.want.S3Buckets[j].Name
```

Or declare a local helper and apply it to both slices?